edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/ApiVersion.cs

```diff
@@ -1,24 +1,27 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Util.Edged
 {
+    using System;
     using System.Collections.Generic;
 
-    public sealed class ApiVersion
+    public sealed class ApiVersion : IComparable<ApiVersion>
     {
+        public static readonly ApiVersion VersionUnknown = new ApiVersion(0, "Unknown");
```

Comment:
I think it shouldn't change the unknown value because it compares the values and gets the lowest, I see that in some places it checks explicitly for unknown, but in SystemResourcesMetrics it just compares the values.

Reply:
Understood. This change actually seems unnecessary for interoperability with `CompareTo >= 0` since `GetVersioned{ModuleManagement,WorkloadClient}` throw exceptions when the parsed version is `VersionUnknown`, so we are guaranteed not to have `this.Value == 100`.